src/codegen_call_dispatch.cpp

```
@@ -84,6 +84,16 @@ llvm::Value *CodeGen::emitExprVariant(const std::unique_ptr<CallExpr> &e) {
         return builder_.CreateCall(getCountFn, {nameStr}, "call_count");
     }
 
+    // Fast path: pre-emit args[0] once for callee names shared by multiple
+    // dispatchers, then route through the same try-chain order to avoid
+    // emitting dead IR from earlier dispatchers that don't match the type.
+    if (e->args.size() >= 2 && (e->callee == "map" || e->callee == "filter")) {
+        llvm::Value *arg0 = emitExpr(*e->args[0]);
```

Clang (14.0.6)

[warning] variable 'arg0' is not initialized

(cppcoreguidelines-init-variables)